### ISSUE TITLE
"sanity check" tile

### ISSUE THREAD

**Issue by Thanood** [monterey-framework/monterey#365]
### Intention

**I'm submitting a feature request**  

**Expected/desired behavior:**  

We could provide a tile or dialog which checks if npm/jspm/au is actually present on `PATH`, prints out some debug information, like `PATH` contents.

**What is the motivation / use case for changing the behavior?**  
:point_up: [4. Oktober 2016 13:14](https://gitter.im/monterey-framework/discussions?at=57f38f283c59573f6f08d771)

**Comment by JeroenVinke** [comment]
What about showing it in the project information tile?

**Comment by Thanood** [comment]
I thought one should be able to open it when no project is available. For example when creating a project already fails because Monterey cannot call `npm install`.

Maybe the support dialog?

**Comment by adriatic** [comment]
One of the relatively strong critique from Rob was "too many tile" - issue that we are already tracking, as I remember. So, adding a sanity check tile might not be the best way to do the sanity check.

Support dialog seems a lot better idea:

![image](https://cloud.githubusercontent.com/assets/2712405/19077525/fe696974-8a1a-11e6-89fa-d33f3ea8bea8.png)

**Comment by Thanood** [comment]
@adriatic I really like your suggestion and changed the title of this issue to reflect that. 😃

**Comment by adriatic** [comment]
@Thanood - it's really easy to enhance a well engineered piece of software like Monterey - you almost cannot do anything bad 😏 

If you are not getting my drift: ___is it kudos to @JeroenVinke___

**Comment by JeroenVinke** [comment]
jspm/au/npm aren't actually on the PATH though. your global node_modules folder is, and that folder contains `npm.cmd`, `jspm.cmd` and `au.cmd`. It would be nice if we were able to detect whether or not something (like aurelia-cli) is installed.

**Comment by Thanood** [comment]
You're right for Windows, but the issue in my original post was on a Linux machine.
Of course we'd need to look for `*.cmd` when doing this.

> It would be nice if we were able to detect whether or not something (like aurelia-cli) is installed.

Even better. But my intention was to make it clearly visible if Monterey can start npm. If it's not on the `PATH` because Linux is strange ( :wink: ) then we'd see it there.

**Comment by adriatic** [comment]
> If it's not on the PATH because Linux is strange ...

It is very common practice to make symbolic links to a folder that is implicitly on the `PATH` than to add something to the `PATH` - we should not fight such established habits. 

We should really think hard and fortify the "sanity" check and make all that code be executed immediately after Monterey installation: Monterey's claim to fame is to make all these "boring" issues related to application building "invisible" to the user - and making sure that Monterey **can run** is the first of these issues.

**Comment by adriatic** [comment]
#### Some ideas for this (sanity check) functionality related to https://github.com/monterey-framework/monterey/issues/368

1. Use [network.js](https://www.npmjs.com/package/network-js) module

**Comment by JeroenVinke** [comment]
https://www.npmjs.com/package/network-js is pretty nice.

**Comment by adriatic** [comment]
We should try to add the `network.js` based network connection measurements to [Verify Installation](https://github.com/monterey-framework/monterey/issues/365#issuecomment-251399451) tab as soon as possible.

**Comment by adriatic** [comment]
See https://github.com/monterey-framework/monterey/issues/386
